leetcodePython/Stack/stack_907.py

```diff
@@ -0,0 +1,19 @@
+from typing import List
+
+
+class Solution:
+    def sumSubarrayMins(self, arr: List[int]) -> int:
+        MOD = 10 ** 9 + 7
+        smaller_element_index = []
+        dp = [0] * len(arr)
+        for i, data in enumerate(arr):
```

Comment:
For each iteration, we try to find `sum of minimums` of all the sub arrays end at `i`